### PR TITLE
Added Language suffix for subtitle downloads

### LINKED ISSUE #9466: Language suffix for subtitle downloads
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

I'd wish for the option of automatically adding a language suffix when downloading subtitles.

### Why do you want this feature?

For many videos, I download multiple subtitles, but as of now, I need to type manually the suffix in the filename. It'd be quicker if it could automatically add one.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by pratyaksh1610**
#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

Added Language suffix for subtitle downloads.

#### Before/After Screenshots/Screen Record
<table>
  <tr>
    <th>Before</th>
    <th>After</th>
  </tr>
  <tr>
    <td>

https://user-images.githubusercontent.com/76740999/205491593-edec6e4a-682c-4e08-92f4-19c3e78f5542.mp4

</td>
    <td>

https://user-images.githubusercontent.com/76740999/205491598-6efdbdbf-240f-440f-bc23-7fa71669b0c6.mp4

</td>
  </tr>
</table>

#### Fixes the following issue(s)

- Fixes #9466 

#### Relies on the following changes

- 

#### APK testing

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by TobiGr — COMMENTED** [review]
Thank you!
One thought about UX: The file name is reset when the user changes the subtitle language. That might be unfortunate when the user already typed in a different file name.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -452,6 +452,7 @@
     <string name="resize_zoom">Zoom</string>
     <string name="caption_auto_generated">Auto-generated</string>
     <!-- Caption Settings -->
+    <string name="caption_file_name">-%s</string>
```
[the remark this answers is not in the document]
Comment (pratyaksh1610) [reply app/src/main/res/values/strings.xml]: Done

File: app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java [hunk 2]
```diff
@@ -562,6 +562,24 @@ public void onItemSelected(final AdapterView<?> parent,
...
+        switch (dialogBinding.videoAudioGroup.getCheckedRadioButtonId()) {
+            case R.id.audio_button:
+            case R.id.video_button:
+                dialogBinding.fileName.setText(FilenameUtils.createFilename(getContext(),
+                        currentInfo.getName()));
+                break;
+            case R.id.subtitle_button:
+                dialogBinding.fileName.setText(FilenameUtils.createFilename(getContext(),
+                currentInfo.getName() + getString(
+                        R.string.caption_file_name, setSubtitleLanguageCode)));
```
[the remark this answers is not in the document]
Comment (pratyaksh1610) [reply app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java]: Done

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -452,6 +452,7 @@
     <string name="resize_zoom">Zoom</string>
     <string name="caption_auto_generated">Auto-generated</string>
     <!-- Caption Settings -->
+    <string name="caption_file_name">%1$s-%2$s</string>
```
Comment (Stypox) [inline app/src/main/res/values/strings.xml:455]: As TobiGr said, make this non-translatable by moving it to `donottranslate.xml`

File: app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java [hunk 4]
```diff
@@ -562,6 +562,24 @@ public void onItemSelected(final AdapterView<?> parent,
                 selectedSubtitleIndex = position;
                 break;
         }
+        onItemSelectedSetFileName();
+    }
+
+    private void onItemSelectedSetFileName() {
+        final String setSubtitleLanguageCode = subtitleStreamsAdapter.getItem(selectedSubtitleIndex)
+                .getLanguageTag();
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java:570]: Move this in between lines 578 and 579 (add `{}` around the `switch` branch if the compiler complains).

File: app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java [hunk 5]
```diff
@@ -562,6 +562,24 @@ public void onItemSelected(final AdapterView<?> parent,
...
+        final String fileName = FilenameUtils.createFilename(getContext(),
+                currentInfo.getName());
+        switch (dialogBinding.videoAudioGroup.getCheckedRadioButtonId()) {
+            case R.id.audio_button:
+            case R.id.video_button:
+                dialogBinding.fileName.setText(fileName);
+                break;
+            case R.id.subtitle_button:
+                dialogBinding.fileName.setText(fileName + getString(
+                        R.string.caption_file_name, fileName, setSubtitleLanguageCode));
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java:580]: What is the `filename + `??
```suggestion
                dialogBinding.fileName.setText(getString(R.string.caption_file_name,
                        fileName, setSubtitleLanguageCode));
```

**Review by Stypox — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (pratyaksh1610) [reply app/src/main/res/values/strings.xml]: Done

File: app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java (hunk 4 above)
Comment (pratyaksh1610) [reply app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java]: Done

**Review by Stypox — APPROVED** [review]
I applied TobiGr's suggestion about UX and pushed a commit. Thank you!

**Pull request merged**